script.js

```diff
@@ -35,25 +35,43 @@
     TOMApp.prototype.render = function() {
       var self = this;
 
+      // reset the region
       self.regions.$mainRegion.empty();
+      $(".action-remove").off("click"); // prevent memory leaking
```

Comment:
For subsequent call to render function, empty the existing region and turn off event handler to address memory leaking.